cyctl/internal/create/template_auth_rules.go

```diff
@@ -0,0 +1,119 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cyclops-ui/cyclops/cyclops-ctrl/api/v1alpha1"
+	"github.com/cyclops-ui/cyclops/cyclops-ctrl/api/v1alpha1/client"
+	"github.com/cyclops-ui/cycops-cyctl/internal/kubeconfig"
+	"github.com/spf13/cobra"
+	v1Spec "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	createTemplateAuthRuleExample = ` 
+	# Create templateauth rule
+	cyctl create templateauthrule NAME --repo='github.com/repo/a' --username='name:key' --password='name:key'
+ 
+	# Create templateauthrule
+	cyctl create templateauthrule demo-templateauthrule --repo='https://github.com/cyclops-ui/templates' --username='name:john' --password='name:random'`
+)
+
+var (
+	username string
+	password string
+)
+
+func validateSecretKeySelector(username, password string) (string, string, string, string, error) {
+	usernameName, usernameKey := _splitNameKey(username)
+	passwordName, passwordKey := _splitNameKey(password)
+
+	// Ensure both name and key are present
+	if usernameName == "" || usernameKey == "" || passwordName == "" || passwordKey == "" {
+		return "", "", "", "", fmt.Errorf("invalid format for username or password. Expected 'name:key'")
+	}
+
+	return usernameName, usernameKey, passwordName, passwordKey, nil
+}
+
+func _splitNameKey(input string) (string, string) {
```

Comment:
@siddhantprateek, how come you added the underscore? The function is already not exported so you can remove it